apps/dashboard/src/pages/environments.tsx

```diff
@@ -1,13 +1,29 @@
 import { PageMeta } from '@/components/page-meta';
+import { ApiServiceLevelEnum } from '@novu/shared';
 import { useEffect } from 'react';
 import { CreateEnvironmentButton } from '../components/create-environment-button';
 import { DashboardLayout } from '../components/dashboard-layout';
+import { FreeTierState } from '../components/environments-free-state';
 import { EnvironmentsList } from '../components/environments-list';
+import { useAuth } from '../context/auth/hooks';
+import { useFetchEnvironments } from '../context/environment/hooks';
+import { useFetchSubscription } from '../hooks/use-fetch-subscription';
 import { useTelemetry } from '../hooks/use-telemetry';
 import { TelemetryEvent } from '../utils/telemetry';
 
 export function EnvironmentsPage() {
+  const { currentOrganization } = useAuth();
+  const { environments = [], areEnvironmentsInitialLoading } = useFetchEnvironments({
+    organizationId: currentOrganization?._id,
+  });
   const track = useTelemetry();
+  const { subscription } = useFetchSubscription();
+
+  const isPaidTier =
+    subscription?.apiServiceLevel === ApiServiceLevelEnum.BUSINESS ||
+    subscription?.apiServiceLevel === ApiServiceLevelEnum.ENTERPRISE;
```

Comment:
This is a more robust check in case we introduce a new paid tier in the future.

```suggestion
  const isPaidTier =
    subscription?.apiServiceLevel !== ApiServiceLevelEnum.FREE;
```